core/pkg/resultshandling/reporter/v2/reporteventreceiver.go

```diff
@@ -202,7 +217,9 @@ func (report *ReportEventReceiver) setResources(reportObj *reporthandlingv2.Post
 			return fmt.Errorf("failed to unmarshal resource '%s', reason: %v", resourceID, err)
 		}
 
-		if *counter+len(r) >= MAX_REPORT_SIZE && len(reportObj.Resources) > 0 {
+		containers := rawResourceContainerHandler(*resource)
```

Comment:
Call this function BEFORE sending resources to the BE (we first collect all the resources we want to send and only then we start sending).
This should happen around [here](https://github.com/kubescape/kubescape/blob/master/core/pkg/resourcehandler/k8sresources.go#L69).